### ISSUE TITLE
Adding Shared Code

### ISSUE THREAD

**Issue by kobi2c** [fbaligand/kibana-enhanced-table#302]
Would it be possible to add "shared code" text field to computed fields UI? 

## Motive
The idea is "Shared code" will be prepended to each expression:
 - **Formula**
 - **Cell computed CSS**
 - **Rows computed filter**
 - **Rows computed CSS**
 
It will make it much easier to maintain the table since there will be one place to update the repeated code ( variables, columns titles, functions and so on). 

## Sample use case

Suppose we have the following expression in each computed field:

```
cssChange(s1) = if ( s1 >= 0 , "color: #480" , "color:#F00" ); cssChange(value)
```

If we want to update the colors, we need to update all the computed fields. 
It is time consuming and error prone and  :(

However with the shared code feature, we can write in the shared code:

```
cssChange(s1) = if ( s1 >= 0 , "color: #480" , "color:#F00" );
```

And in each computed field:

```
cssChange(value)
```

Now it is much easier to change the colors as we need to update the code in one place :)

**Comment by fbaligand** [comment]
This could be an enhancement.
How would you see that feature?
A textarea named "Shared code"?

**Comment by kobi2c** [comment]
Yes, textarea named "Shared code" in "Enhanced Settings" section.
